metagraph/src/annotation/taxonomy/taxo_classifier.cpp

```diff
@@ -0,0 +1,156 @@
+        }
+    }
+    logger->trace("Finished the TaxoClassifier construction in '{}' sec", timer.elapsed());
+}
+
+void TaxoClassifier::update_scores_and_lca(const TaxId start_node,
+                                           const tsl::hopscotch_map<TaxId, uint64_t> &num_kmers_per_node,
+                                           const uint64_t &desired_number_kmers,
+                                           tsl::hopscotch_map<TaxId, uint64_t> &node_scores,
+                                           tsl::hopscotch_set<TaxId> &nodes_already_propagated,
+                                           TaxId &best_lca,
+                                           uint64_t &best_lca_dist_to_root) {
+    if (nodes_already_propagated.count(start_node)) {
+        return;
+    }
+    uint64_t score_from_processed_parents = 0;
+    uint64_t score_from_unprocessed_parents = num_kmers_per_node.at(start_node);
+
+    std::vector<TaxId> processed_parents;
+    std::vector<TaxId> unprocessed_parents;
+
+    TaxId act_node = start_node;
+    unprocessed_parents.push_back(act_node);
+
+    while (act_node != root_node) {
+        act_node = node_parent[act_node];
+        if (!nodes_already_propagated.count(act_node)) {
+            if (num_kmers_per_node.count(act_node)) {
+                score_from_unprocessed_parents += num_kmers_per_node.at(act_node);
+            }
+            unprocessed_parents.push_back(act_node);
+        } else {
+            if (num_kmers_per_node.count(act_node)) {
+                score_from_processed_parents += num_kmers_per_node.at(act_node);
+            }
+            processed_parents.push_back(act_node);
+        }
+    }
+    for (uint64_t i = 0; i < unprocessed_parents.size(); ++i) {
+        TaxId &act_node = unprocessed_parents[i];
+        node_scores[act_node] =
+                score_from_processed_parents + score_from_unprocessed_parents;
+        nodes_already_propagated.insert(act_node);
+
+        uint64_t act_dist_to_root =
+                processed_parents.size() + unprocessed_parents.size() - i;
+        if (node_scores[act_node] >= desired_number_kmers &&
+            act_dist_to_root > best_lca_dist_to_root) {
+            best_lca = act_node;
+            best_lca_dist_to_root = act_dist_to_root;
+        }
+    }
+    for (uint64_t i = 0; i < processed_parents.size(); ++i) {
+        TaxId &act_node = processed_parents[i];
+        node_scores[act_node] += score_from_unprocessed_parents;
+
+        uint64_t act_dist_to_root = processed_parents.size() - i;
+        if (node_scores[act_node] >= desired_number_kmers &&
+            act_dist_to_root > best_lca_dist_to_root) {
+            best_lca = act_node;
+            best_lca_dist_to_root = act_dist_to_root;
+        }
+    }
+}
+
+TaxId TaxoClassifier::assign_class(const mtg::graph::DeBruijnGraph &graph,
+                                   const std::string &sequence,
+                                   const double &lca_coverage_threshold) {
```

Comment:
pass standard types by value instead of const reference.
```suggestion
                                   double lca_coverage_threshold) {
```

Reply:
I changed it to const value, is it ok? Const arguments are colored a little nicer in my IDE. Also, what IDE are you using? My jetbrains doesn't know to link all the files in the project and then "find usages" doesn't work properly. And when you are doing changes in the code, are you working on your machine or directly on the cluster?